File: 4A/ArithmeticMeanVBNET/ArithmeticMeanVBNET/Statistics.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing.Text
Imports System.Text
Imports System.Windows.Forms.VisualStyles

Class Statistics
        Public Property avg As Double = 0.0
        Public Property count As Integer = 0
        Public Property distribution As Dictionary(Of String, Integer) = New Dictionary(Of String, Integer)()

        Public Sub OnlineArithmeticMean(ByVal number As Double)
            Me.count += 1
            Me.avg = (Me.avg * (count - 1) + number) / count
        End Sub

        Public Sub OnlineDistribution(ByVal key As String, ByVal value As Integer)
            If distribution.ContainsKey(key) Then
                distribution(key) += value
            Else
                distribution.Add(key, value)
            End If
        End Sub
End Class

File: 4A/ArithmeticMeanCSharp/ArithmeticMeanCSharp/Statistics.cs
using System;
using System.Collections.Generic;
using System.Drawing.Text;
using System.Text;
using System.Windows.Forms.VisualStyles;

namespace ArithmeticMeanCSharp
{
    class Statistics
    {
        public double avg { get; private set; } = 0.0;
        public int count { get; private set; } = 0;

        public Dictionary<string, int> distribution { get; private set; } = new Dictionary<string, int>();

        public void OnlineArithmeticMean(double number)
        {
            this.count++;
            this.avg = (this.avg * (count - 1) + number) / count;
        }

        public void OnlineDistribution(string key, int value)
        {
            if (distribution.ContainsKey(key))
            {
                distribution[key] += value;
            }
            else
            {
                distribution.Add(key, value);
            }
        }

    }


}
